plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +83,220 @@ protected DistributedQueryRunner createQueryRunner()
             throws Exception
     {
-        return new OperationCounts((Map<OperationContext, Integer>) BEAN_SERVER.getAttribute(new ObjectName(BEAN_NAME), OPERATION_COUNTS_ATTRIBUTE));
+        assertUpdate("CREATE TABLE test_explain AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("EXPLAIN SELECT * FROM test_explain",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
     }
 
-    private static class OperationCounts
+    @Test
+    public void testShowStatsForTable()
+            throws Exception
     {
-        private final Map<OperationContext, Integer> allCounts;
-        private Optional<String> queryId = Optional.empty();
-        private Predicate<String> pathPredicate = path -> true;
-        private Optional<TrackingFileIoProvider.OperationType> type = Optional.empty();
+        assertUpdate("CREATE TABLE test_show_stats AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR test_show_stats",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
 
-        public OperationCounts(Map<OperationContext, Integer> allCounts)
+    @Test
+    public void testShowStatsForTableWithFilter()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_show_stats_with_filter AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR (SELECT * FROM test_show_stats_with_filter where age >= 2)",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
+
+    private void assertFileSystemAccesses(String query, Multiset expectedAccesses)
```

Comment:
raw type usage is intentional here, but still generates a warning, which isn't nice

i would rather not have to ignore warnings, as they are sometimes useful, so would write it as

```
assertThat(ImmutableMultiset.<Object>copyOf(getOperations()))
    .containsExactlyInAnyOrderElementsOf(expectedAccesses);
```

the other way around, ie adapting the expected value, is definitely more verbose. that's the best i could come with:

```
private void assertFileSystemAccesses(String query, Multiset<?> expectedAccesses)
        throws Exception
{
    resetCounts();
    getDistributedQueryRunner().executeWithQueryId(TEST_SESSION, query);
    assertThat(getOperations()).containsExactlyInAnyOrderElementsOf(
            () -> expectedAccesses.stream()
                    .map(FileOperation.class::cast)
                    .iterator());
}
```

Reply:
ok I will go with first approach

Reply:
Note that [assertj-guava](http://joel-costigliola.github.io/assertj/assertj-guava.html) has specific assertions for `Multiset`. I didn't look closely to see if they would be useful here, but check it out.

Reply:
Maybe I missed something but I don't see it to be much more useful. Our concern comes from the fact that java is not able to correctly determine types during creations of Multisets and and these assertions won't help. Please correct me if I'm wrong

Reply:
Maybe the multiset assert would produce better err messages when occurrence counts differ?
when working with the metastore invocations tests, i needed to manually count "GET_TABLE" on the output to adjust the test appropriately.

anyway, here the counts are smaller, so shouldn't be a big deal
or maybe we convert the multiset to counters map for ssertion, and that may produce useful messages without pulling a dependency?

Reply:
> Maybe the multiset assert would produce better err messages when occurrence counts differ?
when working with the metastore invocations tests, i needed to manually count "GET_TABLE" on the output to adjust the test appropriately.

`elements not found:
  <[FileOperation{fileType=METADATA_JSON, operationType=INPUT_FILE_NEW_STREAM},
    FileOperation{fileType=METADATA_JSON, operationType=INPUT_FILE_NEW_STREAM}]>
    
and elements not expected:
  <[FileOperation{fileType=MANIFEST, operationType=INPUT_FILE_NEW_STREAM},
    FileOperation{fileType=MANIFEST, operationType=INPUT_FILE_NEW_STREAM},
    FileOperation{fileType=MANIFEST, operationType=INPUT_FILE_GET_LENGTH},
    FileOperation{fileType=MANIFEST, operationType=INPUT_FILE_GET_LENGTH}]>`

I actually like this format and it is readable for me but if you insist I can switch to counters map.